verification/src/solidity/verifier.rs

```diff
@@ -274,18 +329,73 @@ impl BytecodeWithConstructorArgs {
         Ok(Self {
             bytecode,
             bytes_after_metadata_hash,
+            source: std::marker::PhantomData,
         })
     }
 }
 
+impl Bytecode<CreationTxInput> {
+    /// Extract constructor arguments using the bytecode obtained as a result of local compilation.
+    /// If there are no constructor arguments, returns `Ok(None)`, otherwise returns `Ok`
+    /// with encoded constructor arguments. If the extraction fails, returns `Err`.
+    pub fn constructor_args(
+        &self,
+        compiled_bytecode: &Bytecode<CompilationResult>,
+    ) -> Result<Option<bytes::Bytes>, VerificationError> {
+        if let Some(constructor_args) = self
+            .bytes_after_metadata_hash
+            .strip_prefix(compiled_bytecode.bytes_after_metadata_hash.as_ref())
+        {
+            if constructor_args.is_empty() {
+                Ok(None)
+            } else {
+                Ok(Some(
+                    self.bytes_after_metadata_hash.slice_ref(constructor_args),
+                ))
+            }
+        } else {
+            Err(VerificationError::ExtraDataMismatch(Mismatch::new(
+                compiled_bytecode.bytes_after_metadata_hash.clone().into(),
+                self.bytes_after_metadata_hash.clone().into(),
+            )))
+        }
+    }
+
+    /// Verifies that bytecode and extra data obtained from creation transaction input
+    /// corresponds to the bytecode and extra data obtained from local compilation result.
+    pub fn verify_bytecode_with_extra_data(
+        &self,
+        compiled_bytecode: &Bytecode<CompilationResult>,
+    ) -> Result<(), VerificationError> {
```

Comment:
It seems like an external function, and not a method of the class, may be make this as 
```rust
pub fn verify_bytecode_with_extra_data(
    transaction: &Bytecode<CreationTxInput>,
    compiled: &Bytecode<CompilationResult>,
)
```

Reply:
or may be I wrong?

Reply:
It may be added as external function as well. But I preferred to make it as a method, just because the code seems to be more structured that way, imo

Reply:
I see what you are talking about. Actually, that can be a method on `Verifier` structure and it may even make more sense. However the reason why I've decided to implement it as a method on `Bytecode`, as I want to hide internal structure of `Bytecode` from outside users. So that way internal representation of `Bytecode` only matters for the structure itself